File: 107598019/HW2/Software Architecture_MSQL/usecase/deleteCourse/DeleteCourseImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Software_Architecture.usecase.deleteCourse
{
    public class DeleteCourseImpl : DeleteCourse, DeleteCourseInput
    {
        private CourseRepository courseRepository;

        private string title;

        public DeleteCourseImpl(CourseRepository courseRepository) {
            this.courseRepository = courseRepository;
        }

        public void execute(DeleteCourseInput input, DeleteCourseOutput output)
        {
            output.setMessage(title);
            courseRepository.delete(input.getTitle());
            
        }

        public string getTitle()
        {
            return title;
        }

        public void setTitle(string title)
        {
            this.title = title;
        }
    }
}

File: 107598069/HW2/sa_hw/app/src/main/java/com/example/sa_hw/HW2UseCase/DeleteCourse/DeleteCourseImpl.java
package com.example.sa_hw.HW2UseCase.DeleteCourse;

import com.example.sa_hw.HW2UseCase.CourseRepository;

public class DeleteCourseImpl implements DeleteCourse {

    private CourseRepository courseRepository;

    public DeleteCourseImpl(CourseRepository courseRepository){
        this.courseRepository = courseRepository;
    }

    @Override
    public void execute(DeleteCourseInput input, DeleteCourseOutput output) {
        output.setMessage(input.getCourseID());
        courseRepository.delete(input.getCourseID());
    }
}
